File: TUnit.TestProject/GlobalSetUpCleanUp.cs
namespace TUnit.TestProject;

public static class GlobalSetUpCleanUp
{
    [Before(Assembly)]
    public static void BlahSetUp()
    {
        // Dummy method
    }

    [Before(Assembly)]
    public static void BlahSetUp2()
    {
        // Dummy method
    }

    [After(Assembly)]
    public static Task BlahCleanUp()
    {
        return Task.CompletedTask;
    }
}

File: TUnit.TestProject.FSharp/GlobalSetUpCleanUp.fs
namespace TUnit.TestProject.FSharp

open System.Threading.Tasks
open TUnit.Core

// F# equivalent of GlobalSetUpCleanUp.cs

module GlobalSetUpCleanUp =
    [<Before(HookType.Assembly)>]
    let BlahSetUp() =
        ()

    [<Before(HookType.Assembly)>]
    let BlahSetUp2() =
        ()

    [<After(HookType.Assembly)>]
    let BlahCleanUp() : Task =
        Task.CompletedTask
